components/net/http_loader.rs

```diff
@@ -1109,6 +1110,11 @@ fn http_network_fetch(request: &Request,
                 }
 
                 loop {
+                    if cancellation_listener.lock().unwrap().cancelled() {
+                        *res_body.lock().unwrap() = ResponseBody::Done(vec![]);
```

Comment:
In the context of https://github.com/servo/servo/pull/18676 I think that at this point the response would have been cached already, and such a cancellation would result in any subsequent request for the same url getting the empty cancelled response from the cache. 

One way to solve this would be for the cache to disregard any cached responses that would have been cancelled. Checking for an empty response body might be one way to do so, perhaps a clearer way would be to use `ResponseBody::Empty` here instead, or even introduce a new `ResponseBody::Cancelled`?

Reply:
Having had a look at the spec [spec](https://fetch.spec.whatwg.org/#http-network-fetch), the following can be considered:

1. At step 5 above, if at that point the fetch has already been aborted/terminated, you could return a network error. 
2. If the fetch is terminated/aborted while the fetch worker is already performing step 10, in addition to what is done here and instead of what I suggested earlier, you could set a new 'aborted' flag on the response. The cache could then also ignore any such aborted responses that would be held in it, when asked to construct a response.

Reply:
Let's add the flag to the response and set that here, and use the network error for prior to this point like gterzian suggested.